### PR TITLE
Feature: scrollbar styling

### PR DESCRIPTION
**Pull request by jamesjwarren**
#65

### CODE REVIEW

File: app/less/default/core/scaffolding.less [hunk 1]
```diff
@@ -0,0 +1,18 @@
+//
+// Scaffolding
+// --------------------------------------------------
+
```
Comment (eopa) [inline app/less/default/core/scaffolding.less]: would suggest putting this in a file call `scrollbar.less` rather then using a generic file, which will end up having lots of random styles for various elements inside.

**Comment by eopa** [comment]
@jamesjwarren did you think about using a javascript module to create a fake scrollbar? This would have the advantage of working across many browsers

**Comment by jamesjwarren** [comment]
@eopa I had thought about it but I felt it was better to use CSS rather than implementing JS scrolling, as it would be more performant but if cross browser support for it is crucial then yeh that would be the only option.

**Comment by eopa** [comment]
@jamesjwarren we may have to use it in future projects, and we should always be thinking about cross browser solutions for our styling.

**Comment by jamesjwarren** [comment]
I've trialed a number of JS scrollbar solutions, found this one to be the best but still a little buggy; sometimes the scrollbar remains when there is no content to scroll, it never seems to scroll to the bottom and scrolling is super slow in firefox.

Also, as with any of these, you have to ensure it is aware of changes to content, sometimes it is necessary to manually broadcast/emit the `"content.changed"` event to trigger a redraw of the scrollbar.

A lot more to think about/debug than using native scrollbars.

File: app/index.html [hunk 2]
```diff
@@ -43,8 +43,12 @@
...
 
         <header class="header" ng-include="'partials/header.html'"></header>
-        <div class="sidebar" ng-include="'partials/sidebar.html'" ng-class="{ 'open': sidebarOpen }"></div>
-        <ng-view></ng-view>
+        <div class="sidebar" ng-scrollable>
+            <div ng-include="'partials/sidebar.html'" ng-class="{ 'open': sidebarOpen }"></div>
+        </div>
+        <div class="main-view" ng-scrollable>
+            <ng-view></ng-view>
+        </div>
```
Comment (eopa) [inline app/index.html]: why not just attach the ng-scrollable directive to the existing elements?

This has broken the layout styling for tablet and mobile

File: app/less/default/core/scrollbar.less [hunk 3]
```diff
@@ -0,0 +1,32 @@
+//
+// Scrollbar
+// --------------------------------------------------
+
+@import (css,inline) "@{bower}ng-scrollable/assets/ng-scrollable.css";
```
Comment (eopa) [inline app/less/default/core/scrollbar.less]: You should declare imports in `main.less` so we a top down overview of our less

File: app/less/default/core/variables.less [hunk 4]
```diff
@@ -173,7 +173,7 @@
 
 @sidebar-bg:                    @light-black;
 @sidebar-padding:               30px;
-@sidebar-position-bottom:       @footer-height + (@grid-gutter-width / 2);
+@sidebar-position-bottom:       @footer-height;
```
Comment (eopa) [inline app/less/default/core/variables.less]: this won't leave then leave a small gap between the main elements?

File: app/index.html (hunk 2 above)
Comment (jamesjwarren) [inline app/index.html]: The directive wouldn't work on the existing elements as both `ng-scrollable` `ng-view` and `ng-include` use transclusion.

File: app/less/default/core/variables.less [hunk 5]
```diff
@@ -185,7 +185,7 @@
 
 // ng-view
 @view-padding:                 30px;
-@view-position-bottom:         @footer-height + (@grid-gutter-width / 2);
+@view-position-bottom:         @footer-height;
```
Comment (eopa) [inline app/less/default/core/variables.less]: same comment regarding the above line here as well

File: app/index.html (hunk 2 above)
Comment (eopa) [inline app/index.html]: okay makes sense then, then you just need to fix the styling on mobile and tablet

**Comment by eopa** [comment]
@jamesjwarren your right on FF it does seem very slow, unfortunately. Probably means we can't use this. If there's no better solution then I think we should use your original idea and style the scrollbar for webkit only.

**Comment by jamesjwarren** [comment]
@eopa there are scroll settings that can be configured but no way to target the different browsers, so they would conflict. Also noticed touch on mobile isn't well supported. The latest firefox developer edition has a less obtrusive native scrollbar anyway so presumably that is coming in the relatively near future.

**Comment by eopa** [comment]
@jamesjwarren thanks for looking into a cross browser solution, this will have to do for now

:+1: LGTM!

**Pull request merged**